src/cpp/session/modules/SessionObjectExplorer.R

```diff
@@ -444,6 +453,25 @@
       .rs.explorer.inspectDefault(object, context)
 })
 
+.rs.addFunction("explorer.inspectReticulateObject", function(object,
+                                                             context = .rs.explorer.createContext())
+{
+   children <- NULL
```

Comment:
I think the cost of inspecting all these types could end up being very high if you force marshalling to R (if you ask for the value via `[[` you will). So e.g. a large NumPy matrix will get copied to R here. 

I think the lightest weight way to do this is to call .DollarNames on the Python object. The return value of this will include an attribute named "types" which will be the same length as the .DollarNames result, with types encoded as follows:

https://github.com/rstudio/reticulate/blob/master/src/python.cpp#L1537-L1576

In general I'd keep the display of types very simple (just "function" should be fine).